Geena - geena-panzitta/glp_week10assignment.do

```diff
@@ -0,0 +1,267 @@
+********************************************************************************
+* Econ 490: Week 10
+* Geena Panzitta
+* November 15, 2022
+********************************************************************************
+
+{
+	
+set more off
+clear all
+
+*Change username here
+global username "/Users/geenapanzitta/Library/CloudStorage/Box-Box/"
+
+cd "${username}Econ490_Fall2022/week_10/03_assignment"
+
+global civ_density "01_data/CIV_populationdensity.xlsx"
+global civ_section0 "01_data/CIV_Section_0.dta"
+
+global gps_data "01_data/GPS Data.dta"
+
+global tz_elec_15_clean "01_data/Tz_elec_15_clean.dta"
+global tz_elec_10_clean "01_data/Tz_elec_10_clean.dta"
+
+}
+
+/*******************************************************************************
+Q1
+*******************************************************************************/
+
+{
+
+clear
+import excel "${civ_density}", sheet("Population density") firstrow //import excel file
+keep if substr(NOMCIRCONSCRIPTION,1,11) == "DEPARTEMENT" //keep only departement level entries
+
+replace NOMCIRCONSCRIPTION = substr(NOMCIRCONSCRIPTION,16,.) if substr(NOMCIRCONSCRIPTION,1,15) == "DEPARTEMENT DE " | substr(NOMCIRCONSCRIPTION,1,15) == "DEPARTEMENT D' " | substr(NOMCIRCONSCRIPTION,1,15) == "DEPARTEMENT DU " //take out the beginning of the string
+
+replace NOMCIRCONSCRIPTION = substr(NOMCIRCONSCRIPTION,15,.) if substr(NOMCIRCONSCRIPTION,1,14) == "DEPARTEMENT D'" //take out the beginning of the string
+
+replace NOMCIRCONSCRIPTION = strlower(NOMCIRCONSCRIPTION) //make string lowercase to match merge format
+rename NOMCIRCONSCRIPTION department //rename variable to match other data
+replace department = "arrha" if department == "arrah" //fix spelling inconsistency
+
+tempfile density //save data to tempfile
+save `density'	
+
+clear
+use "${civ_section0}" //load data
+decode b06_departemen, generate(department) //decode to make string variable
+
+merge m:1 department using `density' //merge with density data
+
+}
+
+/*******************************************************************************
+Q2
+*******************************************************************************/
+
+{
+
+clear
+
+tempfile assignments //generate tempfile for enumerator assignments
+save `assignments', emptyok
+
+use "${gps_data}" //import gps data
+egen h_total = count(id) //count number of households
```

Comment:
Instead of generating the h_total variable, saving it as a local, and dropping it, you can use local hn = _N